src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -1,16 +1,29 @@
 package mate.academy;
 
+import java.util.Map;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.Executor;
 
 public class AsyncRequestProcessor {
     private final Executor executor;
+    private final Map<String, UserData> cash = new ConcurrentHashMap<>();
 
     public AsyncRequestProcessor(Executor executor) {
         this.executor = executor;
     }
 
     public CompletableFuture<UserData> processRequest(String userId) {
-        return null;
+        if (cash.containsKey(userId)) {
+            return CompletableFuture.supplyAsync(() -> cash.get(userId));
```

Comment:
If the data is already in the cache, you don't need to use supplyAsync because fetching from the cache is an immediate operation that doesn't need to be executed asynchronously.